File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.go
func maxProfit(prices []int) int {
    low := 10000
    ans := 0

    for _,num := range(prices){

        if(num < low){
        low = num;
        }

        if(num - low > ans){
            ans = num - low
        }
    }
    return ans
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int ans=0;

        int low=Integer.MAX_VALUE;

        for(int i = 0; i<prices.length;i++)
        {
            if(prices[i]<low) low = prices[i];

            if(prices[i] - low > ans) ans = prices[i] - low;
        }
        return ans;
    }
}